src/backend/tests/reimbursement-requests.test.ts

```diff
@@ -222,4 +222,46 @@ describe('Reimbursement Requests', () => {
       expect(prisma.reimbursement_Request.update).toHaveBeenCalledTimes(1);
     });
   });
+
+  describe('Delete Reimbursement Request Tests', () => {
+    test('Request fails when Id does not exist', async () => {
+      jest.spyOn(prisma.reimbursement_Request, 'findUnique').mockResolvedValue(null);
+
+      await expect(() =>
+        ReimbursementRequestService.deleteReimbursementRequest(GiveMeMyMoney.reimbursementRequestId, batman)
+      ).rejects.toThrow(new NotFoundException('Reimbursement Request', GiveMeMyMoney.reimbursementRequestId));
+    });
+
+    test('Request fails if project is already deleted', async () => {
+      jest
+        .spyOn(prisma.reimbursement_Request, 'findUnique')
+        .mockResolvedValue({ ...GiveMeMyMoney, dateDeleted: new Date() });
+      await expect(() =>
+        ReimbursementRequestService.deleteReimbursementRequest(GiveMeMyMoney.reimbursementRequestId, batman)
+      ).rejects.toThrow(new DeletedException('Reimbursement Request', GiveMeMyMoney.reimbursementRequestId));
+    });
+
+    test('Request fails when deleter is not the creator', async () => {
+      jest.spyOn(prisma.reimbursement_Request, 'findUnique').mockResolvedValue(GiveMeMyMoney);
+
+      await expect(() =>
+        ReimbursementRequestService.deleteReimbursementRequest(GiveMeMyMoney.reimbursementRequestId, superman)
+      ).rejects.toThrow(
+        new AccessDeniedException(
+          'You do not have access to delete this reimbursement request, only the creator can delete a reimbursement request'
+        )
+      );
+    });
+
+    test('Request succeeds', async () => {
+      jest.spyOn(prisma.reimbursement_Request, 'findUnique').mockResolvedValue(GiveMeMyMoney);
+      jest.spyOn(prisma.reimbursement_Request, 'update').mockResolvedValue(GiveMeMyMoney);
```

Comment:
dont u need to spread operator and make it so that the date deleted is a date, im suprised that the givememymoney.dateDeleted.isDefined expect is passing rn.